### ISSUE TITLE
Single arity template handlers and rails 6.0.0.beta2

### ISSUE THREAD

**Issue by remomueller** [haml/haml#1007]
I'm getting a "Single arity template handlers" error when using haml with the new rails 6.0.0.beta2.

```
DEPRECATION WARNING: Single arity template handlers are deprecated.  Template handlers must now accept two parameters, the view object and the source for the view object.
Change:
  >> Class#call(template)
To:
  >> Class#call(template, source)
```

It looks like this is caused by this line of code here: https://github.com/haml/haml/blob/c32fa3c3356d3f4c1df73518910f46ea5f5d9fda/lib/haml/plugin.rb#L22

The following approach to fix was used in coffee-rails by Aaron Patterson:

https://github.com/rails/coffee-rails/commit/cbf6af63ac57cea246b75275ba50769d12f43316

I've also seen a similar issue request for the `slim` gem here:
https://github.com/slim-template/slim/issues/827

This is also related to the following rails issue:
https://github.com/rails/rails/issues/35413

Thanks! Let me know if you need additional information!

**Comment by alexvirtualbr** [comment]
Here I'm getting this error above when I run `rails db:migrate`

> DEPRECATION

 WARNING: Single arity template handlers are deprecated.  Template handlers must
now accept two parameters, the view object and the source for the view object.
Change:
  >> Class#call(template)
To:
  >> Class#call(template, source)
 (called from <main> at /myApp/Rakefile:6)

**Comment by cbothner** [comment]
Closed by #1008

**Comment by amatsuda** [comment]
done via #1008

**Comment by nezirz** [comment]
Upgrading the coffee-rails fixed issue in my case on ruby 2.6.4 and rails 6.0.0

**Comment by sreenak** [comment]
i got the same error,i was using  coffee-rails gem version 4.2, updating coffee-rails gem version to 5.0.0 should resolve your problem. ref: https://rubygems.org/gems/coffee-rails/versions/5.0.0
